llama.cpp

```diff
@@ -15574,6 +15574,51 @@ LLAMA_API int32_t llama_chat_apply_template(
     return res;
 }
 
+LLAMA_API int32_t llama_chat_apply_antiprompt(
+              const struct llama_model * model,
+                            const char * tmpl,
+                                  char * buf) {
+    std::string tmpl_str(tmpl == nullptr ? "" : tmpl);
+    if (tmpl == nullptr) {
+        GGML_ASSERT(model != nullptr);
+        // load template from model
+        std::vector<char> model_template(2048, 0); // longest known template is about 1200 bytes
+        std::string template_key = "tokenizer.chat_template";
+        int32_t res = llama_model_meta_val_str(model, template_key.c_str(), model_template.data(), model_template.size());
+        if (res < 0) {
+            // worst case: there is no information about template, we will use chatml by default
+            tmpl_str = "chatml"; // see llama_chat_apply_template_internal
+        } else {
+            tmpl_str = std::string(model_template.data(), model_template.size());
+        }
+    }
+
+    std::string antiprompt;
+
+    if (tmpl_str == "chatml" || tmpl_str.find("<|im_start|>") != std::string::npos) {
```

Comment:
Probably we need to introduce assign one `enum` for each template instead of having to duplicate all this code from `llama_apply_chat_template`